framework/src/base/RandomGenerator.C

```diff
@@ -0,0 +1,45 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "RandomGenerator.h"
+#include  <random>
+
+class MersenneTwister
+{
+public:
+  std::mt19937 _mt_eng;
+};
```

Comment:
1. What is the purpose of this class? Why cannot you put `std::mt19937 _mt_eng` directly into `RandomGenerator`?

2. We are using Mersenne-Twister in `MooseRandom`, in case you are not aware of that class. Maybe you could use it in your `RandomGenerator` class. Or maybe even replace `RandomGenerator` with `MooseRandom`. If `MooseRandom` is a no-go for your needs, I'd like to hear why...